components/sst_importer/src/sst_importer.rs

```diff
@@ -400,7 +400,7 @@ impl ImportDir {
         engine.prepare_sst_for_ingestion(&path.save, &path.clone)?;
         let length = meta.get_length();
         let crc32 = meta.get_crc32();
-        if length != 0 || crc32 != 0 {
+        if length != 0 && crc32 != 0 {
             // we only validate if the length and CRC32 are explicitly provided.
             engine.validate_sst_for_ingestion(cf, &path.clone, length, crc32)?;
```

Comment:
We should validate sst file size.

Reply:
okay, check filesize and crc32 separately, when filesize or crc32 > 0